tests/test_iet.py

```diff
@@ -273,6 +276,59 @@ def _C_free(self):
 }"""
 
 
+def test_cpp_local_object():
+    """
+    Test C++ support for LocalObjects.
+    """
+
+    class MyObject(LocalObject):
+        dtype = CustomDtype('dummy')
+
+    # Locally-scoped objects are declared in the function body
+    lo0 = MyObject('obj0')
+
+    # Globally-scoped objects must not be declared in the function body
+    lo1 = MyObject('obj1', is_global=True)
+
+    # A LocalObject using both a template and a modifier
+    class SpecialObject(LocalObject):
+        dtype = CustomDtype('bar', ('int', 'float'), '&')
+
+    lo2 = SpecialObject('obj2')
+
+    # A LocalObject instantiated and subsequently assigned a value
+    lo3 = MyObject('obj3', initvalue=Macro('meh'))
+
+    # A LocalObject instantiated calling its 2-args constructor and subsequently
+    # assigned a value
+    lo4 = MyObject('obj4', cargs=(1, 2), initvalue=Macro('meh'))
+
+    # A LocalObject with generic sympy exprs used as constructor args
+    expr = sympy.Function('ceil')(FLOAT(Symbol(name='s'))**-1)
```

Comment:
Note: we might wanna create place holder for all those `ceil(type`, `floor(type` .... for cleaner use